Library/Formula/node.rb

```diff
@@ -55,42 +56,50 @@ def install
     system "./configure", *args
     system "make", "install"
 
-    resource("npm").stage libexec/"npm" if build.with? "npm"
-  end
+    if build.with? "npm"
+      resource("npm").stage libexec/"npm_install"
 
-  def post_install
-    return if build.without? "npm"
+      # make sure npm can find node
+      ENV["PATH"] = "#{Formula["node"].prefix}/bin:#{ENV["PATH"]}"
+
+      # set log level temporarily for npm's `make install`
+      ENV["NPM_CONFIG_LOGLEVEL"] = "verbose"
+
+      (libexec/"npm_install").cd { system "./configure", "--prefix=#{libexec}/npm" }
+      (libexec/"npm_install").cd { system "make", "install" }
```

Comment:
Agree on the single ` cd ` block, solid idea.

Not sure about the ` rm_rf `. Mike was concerned in the other thread that we'd be causing trouble for ourselves by removing the stuff in ` libexec `, especially if postinstall failed for some reason and then couldn't be re-run because we'd killed the ` npm ` untarred directory.